### ISSUE TITLE
Ellipsis and quotation marks

### ISSUE THREAD

**Issue by defrilitus** [CanisLupus/swift-selection-search#107]
An almost perfect add-on, made even more awesome by the constraints of web extensions!

Three small bits of nitpicking about the context menu:

1. Is it feasible to add an option whereby the search term is automatically sent the search engine enclosed in quotes if there is a space in the search term (e.g., "John Doe")?
2. Is it possible to remove the ellipsis when it isn’t needed? You will see what I mean when you compare https://imgur.com/kIjb6Vf (no ellipsis needed) with https://imgur.com/k9dbqJc (ellipsis needed). If you check the behaviour of SSS with that of “Look up in Dictionary” (second from top), the difference (in ellipses) may be even clearer..
3. I would also prefer straight quotes to curly ones. I realize that this is an extremely pedantic personal preference, but oddly enough the context-menu illustration on SSS’s AMO page shows straight quotes without an ellipsis: https://imgur.com/ULAUWlr. Something must have changed since then ;-)

Thanks again for providing such an excellent add-on.

**Comment by CanisLupus** [comment]
You are welcome! And thanks! :) Web extensions were really painful in the beginning, indeed. I think the transition was a bit rushed, since many things didn't exist at all. They are slowly getting better. :)

So, for the suggestions:

1. Interesting. Feasible, yes. It would require individually customizable search engines, though. The fact that SSS doesn't have that is something that blocks a lot of possibilities, but it is hard to implement in a way that is not confusing (in the options page). Maybe one day?

2. Unfortunately I can't change anything regarding the ellipsis because showing the selected text in the context menu is a browser feature for web extensions, so it's Firefox's code that controls when and where the ellipsis appears. However, they had a [bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1370228) that caused it to always appear. I believe it was fixed in Firefox 59. In mine (60) it seems okay, as you can see in the images for the next point. Can you confirm your version? :)

3. That was simply a consistency decision based on the default "Search Google for [text]", which uses curly quotes. However, I believe it has been like this in SSS for at least a Long Time™ hehe. The screenshots in AMO actually show curly quotes (you can see they are the same as the Google option) but the difference is tiny.

For comparison, I made a screenshot with curly and then straight ones:

![image](https://user-images.githubusercontent.com/532708/43369508-b10f9b82-9367-11e8-8a25-df7990704772.png)
![image](https://user-images.githubusercontent.com/532708/43369511-bfc7f908-9367-11e8-955f-e8461883b8ed.png)

The difference is there, but yeah... ;)

Cheers,
Daniel

**Comment by defrilitus** [comment]
Thanks for answering so quickly!
1. I'm puzzled that this would require individually customisable search engines, but you are certainly more knowledgeable than I am ;-) It was a feature that the [Context Search](https://addons.mozilla.org/en-US/firefox/addon/context-search/) add-on had, but this of course was not a web extension.
2. I stopped using Firefox about a year ago, as I wasn't prepared to give up all my XUL add-ons, so I am now on Waterfox 56.2.2. However, I now understand why all web extensions that show selected text have ellipses! 
3. I user UserChrome.css to hide the default "Search Google for [text]", so I hadn't noticed. But I believe you :-) And I admit that the whole question is pure pedantry ;-)
In conclusion, thanks again for replying so soon; but, above all, thanks for all the work that must have gone into producing this amazing add-on!

**Comment by CanisLupus** [comment]
You're welcome. I was working a bit on SSS when you posted this, actually. :)

1. Well, to be fair it wouldn't _require_ individually customizable engines if it was a global option (same for every engine), but the problem with that is that some sites simply won't deal with the forced quotes in the same way as for example Google, which tries to get queries that exactly match the quotes. So it would need to exist as an option per search engine, and that's much more complicated with the way SSS is right now.

2. Yeah, I understand, a lot of people stayed for that too. Haha it was a simple bug, but it happens. Unfortunately that version is stuck with it. ;)

3. Yes, the difference is very small. ;) I understand pedantry well, don't worry haha. Curly just for consistency, in this case.

Thank you very much! :D

**Comment by defrilitus** [comment]
I have entered this bug into the [Waterfox Reddit](https://www.reddit.com/r/waterfox/comments/92xgc5/ellipsis_bug_in_firefox_and_waterfox_prior_to/) to see if we can find a solution. In the meantime, thanks for all your help.

**Comment by CanisLupus** [comment]
Good thinking! I wish you luck. You're welcome! :)

**Comment by defrilitus** [comment]
Just a thought: would it be possible to add the an option whereby the search term is automatically sent to the search engine enclosed in quotes if there is a space in the search term (e.g., "John Doe") just for Google Search (https://www.google.com/search?q={searchTerms}), which is probably where it would be most useful? Thanks, as always.

**Comment by CanisLupus** [comment]
Although it makes sense in general, unfortunately I can't assume that the user will want the quotes every time there is a space, even for Google. :) That can make searches yield no results or much less useful results if you search for a sentence and for some reason a word is not in the exact place it should be, or is written incorrectly. Google actually deals with this pretty well even with quotes, but it's not desired in some situations.

**Comment by defrilitus** [comment]
OK, you're the boss ;-)

**Comment by defrilitus** [comment]
I posted the ellipsis problem on the Waterfox Github page and this is the answer I received: https://github.com/MrAlex94/Waterfox/issues/708. At this point, it is too technical for me to decipher, so I will "pass the buck" to you. Thanks in advance.

**Comment by CanisLupus** [comment]
Hey there, I've checked the linked issue and it seems that Mr. Graham simply added a few screenshots of the problem and what it should do instead. I think it's just for reference, for whoever will check the bug later. :) Seems to be on the right track.

**Comment by defrilitus** [comment]
Except that the post referring to SSS is titled "A bugged extension" and concludes "One of the context menu items (i.e., SSS] does use an ellipsis, but that's an extension issue (not Mozilla bug 1370228)".
As I mentioned, this goes above my competence, but if you and Waterfox could communicate on this issue to be sure that we're all on the same page, I'd appreciate it!

**Comment by CanisLupus** [comment]
I'm confident that "A bugged extension" is written as such in contrast with "A legacy extension, not bugged" because legacy (pre-WebExtensions) add-ons didn't have this issue. It doesn't seem to mean that the problem lies with SSS, just that it is bugged by the problem. :)

Furthermore, I believe the "extension issue" is related to the "Search for Image On..." in the last image, which has an ellipsis, not SSS.

Either way, I subscribed to the issue, so if anything comes up I'll be notified and can reply accordingly if necessary. :) For now the issue seems well-documented.

Cheers!

**Comment by defrilitus** [comment]
OK, thanks for taking the time to look into this. I love Waterfox and I love SSS, so I really want to keep  using both! And, as always, I appreciate your quick response times!

**Comment by CanisLupus** [comment]
I'm happy that you like SSS :) You're welcome, I'm definitely not usually this quick with replies, but I'll take credit for these past days hehe ;)

If it turns out that I misunderstood the reply you got in the other issue, I'll be sure to add my two cents.

**Comment by defrilitus** [comment]
👍

**Comment by CanisLupus** [comment]
Hey! :) 

I wanted to tell you that I've created a separate issue for the "quotes around search" suggestion (issue #112) and I'll be closing this one since the other two points you mentioned were "resolved" (kind of!).

I see the other link still has no response, but I'll keep checking.

Cheers!

**Comment by defrilitus** [comment]
Thanks for letting me know!
    
      
        
          
        
        
          To: CanisLupus/swift-selection-search
          <swift-selection-search@noreply.github.com>
          CC: Defrilitus <alessandro@fontana.io>, Author
          <author@noreply.github.com>
